src/main/java/com/zendesk/maxwell/MaxwellUpdateRowsEvent.java

```diff
@@ -48,4 +52,79 @@ public String sqlOperationString() {
 	public String getType() {
 		return "update";
 	}
+
+	private LinkedList<Pair<Row>> filteredRowsBeforeAndAfter;
+	private boolean performedBeforeAndAfterFilter;
+
+	private List<Pair<Row>> filteredRowsBeforeAndAfter() {
+		if ( this.filter == null)
+			return event.getRows();
+
+		if ( performedBeforeAndAfterFilter )
+			return filteredRowsBeforeAndAfter;
+
+		filteredRowsBeforeAndAfter = new LinkedList<>();
+		for ( Pair<Row> p : event.getRows()) {
+			if ( this.filter.matchesRow(this, p.getAfter()) )
+				filteredRowsBeforeAndAfter.add(p);
+		}
+		performedBeforeAndAfterFilter = true;
+		return filteredRowsBeforeAndAfter;
+	}
+
+	@Override
+	public List<RowMap> jsonMaps() {
+		ArrayList<RowMap> list = new ArrayList<>();
+		Object avalue;
+		Object bvalue;
+		for (Iterator<Pair<Row>> p = filteredRowsBeforeAndAfter().iterator(); p.hasNext(); ) {
+			Pair<Row> rowpair = p.next();
+			Row after = rowpair.getAfter();
+			Row before = rowpair.getBefore();
+
+			RowMap rowMap = new RowMap(
+					getType(),
+					getDatabase().getName(),
+					getTable().getName(),
+					getHeader().getTimestamp() / 1000,
+					table.getPKList(),
+					this.getNextBinlogPosition());
+
+			Iterator<Column> aftIter = after.getColumns().iterator();
+			Iterator<Column> befIter = before.getColumns().iterator();
+			Iterator<ColumnDef> defIter = table.getColumnList().iterator();
+			while ( aftIter.hasNext() && defIter.hasNext() && befIter.hasNext() ) {
+				Column a = aftIter.next();
+				ColumnDef d = defIter.next();
+				Column b = befIter.next();
+
+				if ( a instanceof DatetimeColumn) {
+					avalue = ((DatetimeColumn) a).getLongValue();
+				} else {
+					avalue = a.getValue();
+				}
```

Comment:
some DRY stuff here -- maybe move this check to a `valueForJSON(column, value)` function?